src/Components/test/testassets/TestServer/.gitignore

```diff
@@ -0,0 +1 @@
+wwwroot/**/*
```

Comment:
What's the reason for doing all the copying on build, rather than serving the contents from `../BasicTestApp/wwwroot`?

Reply:
+1 for wanting to know the answer.

Reply:
The files in BasicTestApp/wwwroot were being served by `UseBlazorClientSideStaticfiles` files. ServerStartup does not add `UseBlazorClientSideStaticFiles`.

Some tests for blazor server-side and blazor client-side use these files. I didn't want to copy the files as that might get them out of date, so instead I chose to copy them at build time and added the .gitignore file so that they don't show up as modified on every build change.

I could choose to remove the .gitignore and add them to source control. But we might run into issues where they get out of date. Id rather keep the current solution.

Reply:
I was definitely not suggesting we should make static copies of the files in source control. I agree that would get out-of-date very quickly.

I was wondering whether we had the option to serve the files at runtime from `../BasicTestApp/wwwroot`. That would avoid the copying *and* prevents any of us in the future from getting confused about why editing the copies under `TestServer` doesn't take effect.

Do we have that option? I think it would be greatly preferable if we do have it.

Reply:
@SteveSandersonMS We could add an extra file provider to do that I think, but I tried to stay away from that as it moved the complexity from build time to run time and I wanted to keep the services/middleware config as close as possible to our default apps.

My approach was weighted on the fact that we don't change these files or add new files often, which is why I think it strikes a good balance between keeping the runtime code simple and avoid duplicating/checking-in files.

Does that make sense?